sherpa/astro/io/__init__.py

```diff
@@ -675,9 +682,12 @@ def _pack_image(dataset):
         raise IOErr('notimage', dataset.name)
 
     data = {}
-    header = {}
-    if hasattr(dataset, 'header') and isinstance(dataset.header, dict):
-        header = dataset.header.copy()
+
+    # Data2D does not have a header
+    try:
+        header = dataset.header
+    except AttributeError:
+        header = {}
```

Comment:
This will actually have to be `getattr(dataset, "header", {})` but I appreciate the heads up (and I note that other lines in this function already use the `getattr` approach...).